### DISCUSSION TITLE
Getting tags from private quay.io registry

### DISCUSSION THREAD

**Discussion by lauraseidler** [renovatebot/renovate#21513] [Request Help]
### How are you running Renovate?

Self-hosted

### If you're self-hosting Renovate, tell us what version of Renovate you run.

v35.47.2

### If you're self-hosting Renovate, select which platform you are using.

None

### Was this something which used to work for you, and then stopped?

I never saw this working

### Wanted end result.

We're trying to use renovate to update images from a private quay.io registry. Notably, we do not own the organizaion in quay.io from where the images are coming from - we got a username and password to pull images from our vendor, which also allows access to the /v2 API.

### What you tried so far.

Due to a few previous issues and changes from the past (see below), we currently cannot update images from this registry with renovate, as we do not have an access token to use the quay.io /v1 management API, which renovate uses to fetch the list of tags for quay.io images. We've asked our vendor for a token, but it's not possible for us to get one, as quay.ios permission model does not allow restricting those tokens enough.

However, given that this particular image does not have a lot of tags (way below the 1000 that are fetchable with the /v2 API within 20 pages), I was wondering if allowing to choose which API to use for images from quay.io would be an option - defaulting to the current behaviour, but allowing to switch to the /v2 API via configuration (with the previous limitation of incorrect behaviour if there's more than 1000 tags). 

Issue/discussion references:
- https://github.com/renovatebot/renovate/discussions/9159
- https://github.com/renovatebot/renovate/pull/10093
- https://github.com/renovatebot/renovate/discussions/10188

### Relevant debug logs

<details><summary>Logs</summary>

```
Copy/paste the relevant log(s) here, between the starting and ending backticks
```

</details>

**Comment by lmilbaum** [comment]
Were you able to find the proper configuration which renovates images from quay? We are using private repositories on quay.io. 
By the way, I was able to generate a script (curl commands) which extracts the tags by having username and password only.

**Reply by lauraseidler** [reply]
Unfortunately not - but I'm considering submitting a PR and hope it gets accepted, as this is quite blocking for us.

**Reply by lmilbaum** [reply]
BTW, after few days of digging, I've managed to find the proper configuration which worked for me. But, we use a proxy to quay.io.

**Reply by jonaz** [reply]
@lmilbaum please post the anonymized configuration here?

**Reply by lmilbaum** [reply]
```
module.exports = {
  platform: 'gitlab',
  endpoint: 'https://<gitlab server url>/api/v4/'
  hostRules: [
    {
      hostType: "docker",
      matchHost: "<quay.io proxy url>",
      username: process.env.QUAY_USERNAME,
      password: process.env.QUAY_PASSWORD
    }
  ]
};
```
